### PR TITLE
Improves

### PR DESCRIPTION
**Pull request by leonascimento**
`index.js`
- Replace var to const.
- Using arrow function sintax

---

`src/scripts/calendar`
- Create const MAXIMUM_NUMBER_OF_DAYS
- Created the render method to separate of constructor
- Changed loop "for in " to "forEach"
- Using "createDocumentFragment" because not cause page reflow
- Replace this loop `this.nodes.some` to `this.forEach` because this loop
  is better for this context. Some is used just we need to test if some elements
  in array passes the test implemented by the provided function.

---

`src/scripts/datepicker.js`
- Minimize the access on document object
- Created local variables `let options` because it is accessed twice
- Added inline style on element using cssText this is better to apply once
- Created local variable `let options = this.options;`
- Move class DatePickerRange to other file datepickerRange.js using camel case style. Is advisable that always exists just one class per file.

---

`src/scripts/index.js`
- Import and define like public module.

:bowtie: :sunglasses: :metal:

### CODE REVIEW

File: src/scripts/datepickerRange.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { DatePicker } from './datepicker';
+
+export class DatePickerRange {
+
+    constructor(options) {
+        this.rightDateSelected = false;
+        this.options = options || {
```
Comment (evangilo) [inline src/scripts/datepickerRange.js]: The default values are defined in the class DatePicker. Only leftSelector and rightSelector are required.

File: src/scripts/calendar.js [hunk 2]
```diff
@@ -7,21 +7,31 @@ import {
...
-            this.container.appendChild(this.nodes[i]);
+        this.render();
+    }
+
+    render() {
+        let container = this.container;
+        let nodes = this.nodes;
+        this.weekdays.forEach(weekday => container.appendChild(createNode('span', 'weekday', weekday)));
+
+        let fragment = document.createDocumentFragment();
```
Comment (evangilo) [inline src/scripts/calendar.js:28]: :+1:

File: LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (evangilo) [inline LICENSE.md:1]: :+1:

File: src/scripts/datepicker.js [hunk 4]
```diff
@@ -70,9 +73,7 @@ export class DatePicker {
     show() {
         let parentOffset = this.parentElement.getBoundingClientRect();
         this.isOpened = true;
-        this.datepicker.style.display = 'block';
-        this.datepicker.style.top = `${parentOffset.top + 55}px`;
-        this.datepicker.style.left = `${parentOffset.left}px`;
+        this.datepicker.style.cssText = `display: block; top: ${(parentOffset.top + 55)}px; left: ${parentOffset.left}px;`;
```
Comment (evangilo) [inline src/scripts/datepicker.js:76]: :+1:

**Comment by diasbruno** [comment]
@leonascimento this branch is outdate with master. Must be rebased.

**Comment by diasbruno** [comment]
@leonascimento this PR covers too many features and chores.

File: src/scripts/datepicker.js (hunk 4 above)
Comment (diasbruno) [reply src/scripts/datepicker.js:76]: `cssText` is a good hack,  but terrible to read. :)

File: src/scripts/datepicker.js [hunk 5]
```diff
@@ -84,41 +85,46 @@ export class DatePicker {
     }
 
     selectDate(date) {
-        if (isSame(date, this.options.currentMonth, 'year month')) {
-            this.options.currentDate = date;
+        let options = this.options;
+        if (isSame(date, options.currentMonth, 'year month')) {
+            options.currentDate = date;
```
Comment (diasbruno) [inline src/scripts/datepicker.js:90]: `let options = this.options;` prefer the previous inlined way.

Comment (diasbruno) [inline src/scripts/datepicker.js:90]: Forget that. ^^

**Comment by diasbruno** [comment]
`src/assets/datepicker.gif` is not a source file.

File: src/scripts/datepicker.js (hunk 4 above)
Comment (leonascimento) [reply src/scripts/datepicker.js:76]: @diasbruno @evangilo  Ok, but you must minimise the repaint and reflow and this line just have three styles declarations.

**Pull request merged**